src/generic.rs

```diff
@@ -47,7 +47,12 @@ fn parse_snmp_v3_pdu_content(i: &[u8]) -> IResult<&[u8], SnmpV3Message, SnmpErro
     Ok((i, msg))
 }
 
-pub fn parse_snmp_generic_message(i: &[u8]) -> IResult<&[u8], SnmpGenericMessage, SnmpError> {
+/// Parse any valid SNMP message.
+pub fn parse_snmp_generic_message(i: &[u8]) -> Result<(&[u8], SnmpGenericMessage), SnmpError> {
+    internal_parse_snmp_generic_message(i).finish()
+}
```

Comment:
I understand why merging error types with `.finish()` may seem interesting, but the problem is that it prevents chaining the parser combinators (for ex, you won't be able to use `many1(parse_snmp_generic_message)`.
Another problem is that it breaks backwards API.

Since the new API in `asn1-rs` is more focused around `FromBer`, this is maybe an intermediate solution: providing helper functions (or just let caller add `.finish()` ...) with different names (to not break API), in addition to the default ones